tests/unit/test_python_evaluation.py

```diff
@@ -95,6 +98,52 @@ def python_data():
     return rating_true, rating_pred, rating_nohit
 
 
+def test_python_datatypes(python_data):
+    rating_true, rating_pred, _ = python_data
+
+    # Change data types of true and prediction data, and there should type error produced.
+    rating_true_copy = rating_true.copy()
+
+    rating_true_copy[DEFAULT_USER_COL] = rating_true_copy[DEFAULT_USER_COL].astype(str)
+    rating_true_copy[DEFAULT_RATING_COL] = rating_true_copy[DEFAULT_RATING_COL].astype(str)
+
+    with pytest.raises(TypeError) as e_info:
+        _merge_rating_true_pred(
+            rating_true_copy,
+            rating_pred,
+            col_user=DEFAULT_USER_COL,
+            col_item=DEFAULT_ITEM_COL,
+            col_rating=DEFAULT_RATING_COL,
+            col_prediction=PREDICTION_COL,
+            relevancy_method="top_k",
+        )
+        assert str(e_info.value) == "Data types of column {} are different in true and prediction".format(DEFAULT_USER_COL)
+
+    with pytest.raises(TypeError) as e_info:
+        _merge_rating_true_pred(
+            rating_true_copy,
+            rating_pred,
+            col_user=DEFAULT_USER_COL,
+            col_item=DEFAULT_ITEM_COL,
+            col_rating=DEFAULT_RATING_COL,
+            col_prediction=PREDICTION_COL,
+            relevancy_method="top_k",
+        )
+        assert str(e_info.value) == "Data types of column {} and {} are different in true and prediction".format(DEFAULT_RATING_COL, PREDICTION_COL)
+
```

Comment:
this test is the same as the previous lines except the asserting the error message.